tests/integration/individual_response/test_individual_response.py

```diff
@@ -0,0 +1,375 @@
+        self._add_household_no_primary()
+        self.get(self.individual_section_link)
+        self.get(self.individual_response_link)
+        self.post()
+        self.post({"individual-response-how-answer": "Post"})
+
+        # And I post with no data
+        self.post()
+
+        # Then I should see errors rendered correctly
+        self.assertInUrl("/confirm-address")
+        self.assertInBody("This page has an error")
+
+    def test_display_address_on_confirmation_page(self):
+        # Given I navigate to the confirmation page
+        self._add_household_no_primary()
+        self.get(self.individual_section_link)
+        self.get(self.individual_response_link)
+        self.post()
+        self.post({"individual-response-how-answer": "Post"})
+
+        # And I post with no data
+        self.post(
+            {
+                "individual-response-post-confirm-answer": "Yes, send the access code by post"
+            }
+        )
+
+        # Then I should see errors rendered correctly
+        self.assertInUrl("/confirmation")
+        self.assertInBody("68 Abingdon Road, Goathill")
+
+    def test_individual_response_requested_status(self):
+        # Given I request an individual response by post
+        self._request_individual_response()
+
+        # When I navigate to the hub
+        self.get("/questionnaire")
+
+        # Then I should see "Separate census requested" as
+        # the individual section status
+        self.assertInBody("Separate census requested")
+
+    def test_individual_response_not_requested_status_unchanged(self):
+        # Given I naviagate to the confirm page of individual response
+        # but don't request one
+        self._add_household_no_primary()
+        self.post()
+        self.get(self.individual_response_link)
+        self.post()
+        self.post({"individual-response-how-answer": "Post"})
+
+        # When I navigate to the hub
+        self.get("/questionnaire")
+
+        # Then I should see "Not started" as
+        # the individual section status
+        self.assertInBody("Not started")
+
+    def test_individual_response_page_404_if_primary_id_used(self):
+        # Given I add a primary person
+        self._add_primary()
+
+        # When I navigate to the how endpoint using the primary person's
+        # list item id
+        self.post()
+
+        primary_person_id = self.last_url.split("/")[3]
+        self.get(f"individual-response?list_item_id={primary_person_id}")
+        self.assertInBody("Page not found")
+
+    def test_individual_response_how_page_404_if_primary_id_used(self):
+        # Given I add a primary person
+        self._add_primary()
+
+        # When I navigate to the how endpoint using the primary person's
+        # list item id
+        self.post()
+
+        primary_person_id = self.last_url.split("/")[3]
+        self.get(f"individual-response/{primary_person_id}/how")
+        self.assertInBody("Page not found")
```

Comment:
I think this test file needs to be give a once over for consistency of the Given When Then comments. Here the When should be after the `self.post()` and there should be a Then before the assert.